benchmarks/dynamo/common.py

```diff
@@ -1429,34 +1383,144 @@ def run_with_iobinding(self, iobinding, outputs):
         self.onnx_session.run_with_iobinding(iobinding)
         return outputs
 
+    def run_with_onnx_inputs(self, onnx_inputs):
+        return self.onnx_session.run(None, onnx_inputs)
+
+    @classmethod
+    def save_tensor_data(cls, numpy_tensor, output_path):
+        from onnx import numpy_helper
+
+        proto_tensor = numpy_helper.from_array(numpy_tensor)
+        with open(output_path, "wb") as f:
+            f.write(proto_tensor.SerializeToString())
+
+    def run_and_serialize_inputs_outputs(self, pt_inputs):
+        onnx_inputs = self.adapt_pt_inputs_to_onnx(pt_inputs)
+        onnx_outputs = self.run_with_onnx_inputs(onnx_inputs)
+
+        test_data_dir = self.model_dir / "test_data_set_0"
+        test_data_dir.mkdir(parents=True, exist_ok=True)
+
+        for i, onnx_input in enumerate(onnx_inputs.values()):
+            self.save_tensor_data(onnx_input, str(test_data_dir / f"input_{i}.pb"))
+        for i, onnx_output in enumerate(onnx_outputs):
+            self.save_tensor_data(onnx_output, str(test_data_dir / f"output_{i}.pb"))
+
+        return self.adapt_onnx_outputs_to_pt(onnx_outputs)
+
     def run(self, pt_inputs):
         # NOTE: For CUDA performance testing, use `run_with_iobinding` to exclude memory
         # copying overhead for inputs/outputs between cpu and gpu.
         # Otherwise perf number is inaccurate.
-        pt_inputs = self.format_pt_inputs(pt_inputs)
-        onnx_inputs = {
-            ort_input.name: pt_input.cpu().numpy()
-            for ort_input, pt_input in zip(self.onnx_session.get_inputs(), pt_inputs)
-        }
-        ort_outputs = self.onnx_session.run(None, onnx_inputs)
-        pt_outputs = [
-            torch.from_numpy(ort_output).to(current_device)
-            for ort_output in ort_outputs
-        ]
-        if len(pt_outputs) == 1:
-            return pt_outputs[0]
+        onnx_inputs = self.adapt_pt_inputs_to_onnx(pt_inputs)
+        onnx_outputs = self.run_with_onnx_inputs(onnx_inputs)
+        return self.adapt_onnx_outputs_to_pt(onnx_outputs)
+
+
+class OnnxModelFromTorchScript(OnnxModel):
+    """TorchScript based onnx export. `torch.onnx.export`
+
+    TODO(bowbao):
+    * large model export failed.
+          Onnx Model is larger than 2GB, but exporter makes decision based pt model size, which is
+          smaller than 2GB.
+    * OOM on slightly larger model.
+          Both pt model and ort inference session are on gpu. Attempt has been made to move ORT to
+          cuda:1, however ORT perf drop significantly.
+          For now running everything with batch_size 1 set in launch script.
+    """
+
+    TORCH_TO_NUMPY_DTYPE = {
+        torch.float16: np.float16,
+        torch.float32: np.float32,
+        torch.float64: np.float64,
+        torch.uint8: np.uint8,
+        torch.int8: np.int8,
+        torch.int16: np.int16,
+        torch.int32: np.int32,
+        torch.int64: np.longlong,
+        torch.bool: np.bool_,
+    }
```

Comment:
this enum already exists at https://github.com/pytorch/pytorch/blob/main/torch/onnx/_internal/fx/type_utils.py#L166